plugin/src/main/java/io/jenkins/plugins/casc/ConfigurationAsCode.java

```diff
@@ -558,6 +558,10 @@ private void configureWith(List<YamlSource> sources) throws ConfiguratorExceptio
             throw new ConfiguratorException("Invalid configuration: '"+path+"' isn't a valid path.");
         }
 
+        if (Files.isRegularFile(root)) {
```

Comment:
Minor nitpick could you move the `final PathMatcher matcher = FileSystems.getDefault().getPathMatcher(YAML_FILES_PATTERN);` below this if statement?

Reply:
Done. Also added a check to ensure its readable as well before returning it, only to IOException later.